Libraries/Opc.Ua.PubSub/Transport/UdpClientCreator.cs

```diff
@@ -148,10 +148,14 @@ internal static List<UdpClient> GetUdpClients(UsedInContext pubSubContext, Netwo
             foreach (NetworkInterface nic in usableNetworkInterfaces)
             {
                 Utils.Trace(Utils.TraceMasks.Information, "NetworkInterface name('{0}') attempts to create instance of UdpClient.", nic.Name);
-                //ignore loop-back interface
-                if (nic.NetworkInterfaceType == NetworkInterfaceType.Loopback) continue;
-                //ignore tunnel interface
-                if (nic.NetworkInterfaceType == NetworkInterfaceType.Tunnel) continue;
+
+                //ignore loop-back or tunnel  interfaces
+                bool ignore = nic.NetworkInterfaceType == NetworkInterfaceType.Loopback || nic.NetworkInterfaceType == NetworkInterfaceType.Tunnel;
```

Comment:
given that you added the check for OperationStatus (line 157), wouldn't it be better that the previous checks remain unchanged (without `bool ignore`).

Reply:
That is according to clean code but if you do not practice clean coding then you are right it could remain as it was.